src/components/TopNav.jsx

```diff
@@ -0,0 +1,206 @@
+import React, { Component } from 'react';
+import { NavLink } from 'react-router-dom';
+import { createArticleIcon, notificationIcon } from '../assets/img__func/icons_svg';
+import Logo from './logo';
+import UserNavAvatar from './userNavAvartar';
+// import Hamburger from './Hamburger';
+
+/**
+ * @description top nav
+ * @returns {JSX} top nav
+ */
+class TopNav extends Component {
+  /**
+   * @description top nav contructor
+   * @param {object} props
+   * @returns {undefined}
+   */
+  constructor(props) {
+    super(props);
+    this.state = {
+      mobile: 768,
+      display: '',
+      auth: false,
+      hamburgerStateClose: 'inline-block',
+      hamburgerStateOpen: 'none',
+      reset: false,
+      menuClassStyleName: 'link-wrapper'
+    };
+    this.renderNavChildren = this.renderNavChildren.bind(this);
+    this.renderHamburgerMenu = this.renderHamburgerMenu.bind(this);
+  }
+
```

Comment:
@shaolinmkz, the use of arrow function has taken care of the need to use `.bind` in our react classes this tutorial by Cory house will comee in handy `https://medium.freecodecamp.org/react-binding-patterns-5-approaches-for-handling-this-92c651b5af56` or `https://medium.com/@nikolalsvk/loosing-bind-this-in-react-8637ebf372cf`

Reply:
On it..